BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/MainActivity.java

```diff
@@ -42,36 +58,39 @@ public void onSuccess(LoginResult loginResult) {
                     token.enqueue(loginCallback);
                 } catch (IOException e) {
                     Log.e(TAG, e.toString());
-                    // TODO: https://github.com/uwblueprint/foe/issues/16
+                    showErrorDialog(getString(R.string.error_message));
                 }
             }
 
             @Override
-            public void onCancel() {
-                // TODO: https://github.com/uwblueprint/foe/issues/16
-            }
+            public void onCancel() {}
 
             @Override
             public void onError(FacebookException exception) {
-                // TODO: https://github.com/uwblueprint/foe/issues/16
+                Log.e(TAG, "There was an error on Facebook login. " + exception.toString());
+                showErrorDialog(getString(R.string.error_message));
             }
         });
 
 
         loginCallback = new Callback<BeeTrackerCaller.SignupResponse>() {
             @Override
             public void onResponse(Call<BeeTrackerCaller.SignupResponse> call, Response<BeeTrackerCaller.SignupResponse> response) {
-                Log.d(TAG, "got response string: " + response.body().getToken());
-                // TODO: Store returned token https://github.com/uwblueprint/foe/issues/17
-                Intent intent = new Intent(getApplicationContext(), HomePageActivity.class);
-                startActivity(intent);
-                finish();
+                if (response.code() == 401 || response.body() == null ||  response.body().getToken() == null) {
+                    Log.e(TAG, "response from server is 401 + " + response.message());
+                    showErrorDialog(getString(R.string.error_message));
+                    LoginManager.getInstance().logOut();
+                    return;
+                }
+                SharedPreferences sharedPref = getPreferences(Context.MODE_PRIVATE);
```

Comment:
Can considering storing it more securely: https://stackoverflow.com/questions/43629251/how-to-save-oauth-access-token-securely-in-android